### ISSUE TITLE
The server worker successfully loads the model to the node GPU, but it shows failure

### ISSUE THREAD

**Issue by benchiang** [gpustack/gpustack#1329]
I registered two workers through gpustack, ai-server-34 and ai-server-33, where 34 is the server worker. Now I publish the model to 33 through gpustack on 34. There is no error in the model running log, but the model list reports an error. At the same time, I found that the model has been loaded into the GPU on node 33. The specific running information is as follows:

2025-02-25T06:08:03+00:00 - gpustack.worker.backends.vllm - INFO - Starting vllm server
INFO 02-25 06:08:06 __init__.py:190] Automatically detected platform cuda.
INFO 02-25 06:08:07 api_server.py:840] vLLM API server version 0.7.2
INFO 02-25 06:08:07 api_server.py:841] args: Namespace(subparser='serve', model_tag='/models/ChatGLM4-9B-128K', config='', host='0.0.0.0', port=40286, uvicorn_log_level='info', allow_credentials=False, allowed_origins=['*'], allowed_methods=['*'], allowed_headers=['*'], api_key=None, lora_modules=None, prompt_adapters=None, chat_template=None, chat_template_content_format='auto', response_role='assistant', ssl_keyfile=None, ssl_certfile=None, ssl_ca_certs=None, ssl_cert_reqs=0, root_path=None, middleware=[], return_tokens_as_token_ids=False, disable_frontend_multiprocessing=False, enable_request_id_headers=False, enable_auto_tool_choice=False, enable_reasoning=False, reasoning_parser=None, tool_call_parser=None, tool_parser_plugin='', model='/models/ChatGLM4-9B-128K', task='auto', tokenizer=None, skip_tokenizer_init=False, revision=None, code_revision=None, tokenizer_revision=None, tokenizer_mode='auto', trust_remote_code=True, allowed_local_media_path=None, download_dir=None, load_format='auto', config_format=<ConfigFormat.AUTO: 'auto'>, dtype='auto', kv_cache_dtype='auto', max_model_len=8192, guided_decoding_backend='xgrammar', logits_processor_pattern=None, model_impl='auto', distributed_executor_backend=None, pipeline_parallel_size=1, tensor_parallel_size=1, max_parallel_loading_workers=None, ray_workers_use_nsight=False, block_size=None, enable_prefix_caching=None, disable_sliding_window=False, use_v2_block_manager=True, num_lookahead_slots=0, seed=0, swap_space=4, cpu_offload_gb=0, gpu_memory_utilization=0.9, num_gpu_blocks_override=None, max_num_batched_tokens=None, max_num_seqs=None, max_logprobs=20, disable_log_stats=False, quantization=None, rope_scaling=None, rope_theta=None, hf_overrides=None, enforce_eager=True, max_seq_len_to_capture=8192, disable_custom_all_reduce=False, tokenizer_pool_size=0, tokenizer_pool_type='ray', tokenizer_pool_extra_config=None, limit_mm_per_prompt=None, mm_processor_kwargs=None, disable_mm_preprocessor_cache=False, enable_lora=False, enable_lora_bias=False, max_loras=1, max_lora_rank=16, lora_extra_vocab_size=256, lora_dtype='auto', long_lora_scaling_factors=None, max_cpu_loras=None, fully_sharded_loras=False, enable_prompt_adapter=False, max_prompt_adapters=1, max_prompt_adapter_token=0, device='auto', num_scheduler_steps=1, multi_step_stream_outputs=True, scheduler_delay_factor=0.0, enable_chunked_prefill=None, speculative_model=None, speculative_model_quantization=None, num_speculative_tokens=None, speculative_disable_mqa_scorer=False, speculative_draft_tensor_parallel_size=None, speculative_max_model_len=None, speculative_disable_by_batch_size=None, ngram_prompt_lookup_max=None, ngram_prompt_lookup_min=None, spec_decoding_acceptance_method='rejection_sampler', typical_acceptance_sampler_posterior_threshold=None, typical_acceptance_sampler_posterior_alpha=None, disable_logprobs_during_spec_decoding=None, model_loader_extra_config=None, ignore_patterns=[], preemption_mode=None, served_model_name=['ChatGLM4-9B'], qlora_adapter_name_or_path=None, otlp_traces_endpoint=None, collect_detailed_traces=None, disable_async_output_proc=False, scheduling_policy='fcfs', override_neuron_config=None, override_pooler_config=None, compilation_config=None, kv_transfer_config=None, worker_cls='auto', generation_config=None, override_generation_config=None, enable_sleep_mode=False, calculate_kv_scales=False, disable_log_requests=False, max_log_len=None, disable_fastapi_docs=False, enable_prompt_tokens_details=False, dispatch_function=<function serve at 0x735ed01aae80>)
INFO 02-25 06:08:07 api_server.py:206] Started engine process with PID 7465
INFO 02-25 06:08:11 __init__.py:190] Automatically detected platform cuda.
INFO 02-25 06:08:13 config.py:542] This model supports multiple tasks: {'embed', 'score', 'reward', 'classify', 'generate'}. Defaulting to 'generate'.
WARNING 02-25 06:08:13 cuda.py:95] To see benefits of async output processing, enable CUDA graph. Since, enforce-eager is enabled, async output processor cannot be used
WARNING 02-25 06:08:13 config.py:678] Async output processing is not supported on the current platform type cuda.
WARNING 02-25 06:08:13 tokenizer.py:221] Using a slow tokenizer. This might cause a significant slowdown. Consider using a fast tokenizer instead.
INFO 02-25 06:08:17 config.py:542] This model supports multiple tasks: {'embed', 'score', 'reward', 'generate', 'classify'}. Defaulting to 'generate'.
WARNING 02-25 06:08:17 cuda.py:95] To see benefits of async output processing, enable CUDA graph. Since, enforce-eager is enabled, async output processor cannot be used
WARNING 02-25 06:08:17 config.py:678] Async output processing is not supported on the current platform type cuda.
INFO 02-25 06:08:17 llm_engine.py:234] Initializing a V0 LLM engine (v0.7.2) with config: model='/models/ChatGLM4-9B-128K', speculative_config=None, tokenizer='/models/ChatGLM4-9B-128K', skip_tokenizer_init=False, tokenizer_mode=auto, revision=None, override_neuron_config=None, tokenizer_revision=None, trust_remote_code=True, dtype=torch.bfloat16, max_seq_len=8192, download_dir=None, load_format=LoadFormat.AUTO, tensor_parallel_size=1, pipeline_parallel_size=1, disable_custom_all_reduce=False, quantization=None, enforce_eager=True, kv_cache_dtype=auto,  device_config=cuda, decoding_config=DecodingConfig(guided_decoding_backend='xgrammar'), observability_config=ObservabilityConfig(otlp_traces_endpoint=None, collect_model_forward_time=False, collect_model_execute_time=False), seed=0, served_model_name=ChatGLM4-9B, num_scheduler_steps=1, multi_step_stream_outputs=True, enable_prefix_caching=False, chunked_prefill_enabled=False, use_async_output_proc=False, disable_mm_preprocessor_cache=False, mm_processor_kwargs=None, pooler_config=None, compilation_config={"splitting_ops":[],"compile_sizes":[],"cudagraph_capture_sizes":[],"max_capture_size":0}, use_cached_outputs=True, 
WARNING 02-25 06:08:17 tokenizer.py:221] Using a slow tokenizer. This might cause a significant slowdown. Consider using a fast tokenizer instead.
INFO 02-25 06:08:18 cuda.py:230] Using Flash Attention backend.
INFO 02-25 06:08:18 model_runner.py:1110] Starting to load model /models/ChatGLM4-9B-128K...
Loading safetensors checkpoint shards:   0% Completed | 0/10 [00:00<?, ?it/s]
Loading safetensors checkpoint shards:  10% Completed | 1/10 [00:00<00:02,  3.86it/s]
Loading safetensors checkpoint shards:  20% Completed | 2/10 [00:00<00:02,  3.25it/s]
Loading safetensors checkpoint shards:  30% Completed | 3/10 [00:00<00:02,  3.07it/s]
Loading safetensors checkpoint shards:  40% Completed | 4/10 [00:01<00:01,  3.06it/s]
Loading safetensors checkpoint shards:  50% Completed | 5/10 [00:01<00:01,  3.03it/s]
Loading safetensors checkpoint shards:  60% Completed | 6/10 [00:01<00:01,  3.06it/s]
Loading safetensors checkpoint shards:  70% Completed | 7/10 [00:02<00:00,  3.09it/s]
Loading safetensors checkpoint shards:  80% Completed | 8/10 [00:02<00:00,  3.05it/s]
Loading safetensors checkpoint shards:  90% Completed | 9/10 [00:02<00:00,  3.01it/s]
Loading safetensors checkpoint shards: 100% Completed | 10/10 [00:03<00:00,  2.99it/s]
Loading safetensors checkpoint shards: 100% Completed | 10/10 [00:03<00:00,  3.06it/s]
INFO 02-25 06:08:22 model_runner.py:1115] Loading model weights took 17.5635 GB
INFO 02-25 06:08:23 worker.py:267] Memory profiling takes 0.97 seconds
INFO 02-25 06:08:23 worker.py:267] the current vLLM instance can use total_gpu_memory (79.25GiB) x gpu_memory_utilization (0.90) = 71.33GiB
INFO 02-25 06:08:23 worker.py:267] model weights take 17.56GiB; non_torch_memory takes 0.09GiB; PyTorch activation peak memory takes 1.44GiB; the rest of the memory reserved for KV Cache is 52.23GiB.
INFO 02-25 06:08:23 executor_base.py:110] # CUDA blocks: 85572, # CPU blocks: 6553
INFO 02-25 06:08:23 executor_base.py:115] Maximum concurrency for 8192 tokens per request: 167.13x
INFO 02-25 06:08:25 llm_engine.py:431] init engine (profile, create kv cache, warmup model) took 3.07 seconds
INFO 02-25 06:08:26 api_server.py:756] Using supplied chat template:
INFO 02-25 06:08:26 api_server.py:756] None
INFO 02-25 06:08:26 launcher.py:21] Available routes are:
INFO 02-25 06:08:26 launcher.py:29] Route: /openapi.json, Methods: HEAD, GET
INFO 02-25 06:08:26 launcher.py:29] Route: /docs, Methods: HEAD, GET
INFO 02-25 06:08:26 launcher.py:29] Route: /docs/oauth2-redirect, Methods: HEAD, GET
INFO 02-25 06:08:26 launcher.py:29] Route: /redoc, Methods: HEAD, GET
INFO 02-25 06:08:26 launcher.py:29] Route: /health, Methods: GET
INFO 02-25 06:08:26 launcher.py:29] Route: /ping, Methods: GET, POST
INFO 02-25 06:08:26 launcher.py:29] Route: /tokenize, Methods: POST
INFO 02-25 06:08:26 launcher.py:29] Route: /detokenize, Methods: POST
INFO 02-25 06:08:26 launcher.py:29] Route: /v1/models, Methods: GET
INFO 02-25 06:08:26 launcher.py:29] Route: /version, Methods: GET
INFO 02-25 06:08:26 launcher.py:29] Route: /v1/chat/completions, Methods: POST
INFO 02-25 06:08:26 launcher.py:29] Route: /v1/completions, Methods: POST
INFO 02-25 06:08:26 launcher.py:29] Route: /v1/embeddings, Methods: POST
INFO 02-25 06:08:26 launcher.py:29] Route: /pooling, Methods: POST
INFO 02-25 06:08:26 launcher.py:29] Route: /score, Methods: POST
INFO 02-25 06:08:26 launcher.py:29] Route: /v1/score, Methods: POST
INFO 02-25 06:08:26 launcher.py:29] Route: /rerank, Methods: POST
INFO 02-25 06:08:26 launcher.py:29] Route: /v1/rerank, Methods: POST
INFO 02-25 06:08:26 launcher.py:29] Route: /v2/rerank, Methods: POST
INFO 02-25 06:08:26 launcher.py:29] Route: /invocations, Methods: POST
INFO:     Started server process [7329]
INFO:     Waiting for application startup.
INFO:     Application startup complete.
INFO:     Uvicorn running on http://0.0.0.0:40286 (Press CTRL+C to quit)
INFO:     127.0.0.1:50092 - "GET /v1/models HTTP/1.1" 200 OK

<img width="239" alt="Image" src="https://github.com/user-attachments/assets/063213fa-005e-4514-8b70-3f5779b0f9d2" />

<img width="1092" alt="Image" src="https://github.com/user-attachments/assets/19d9dcf9-7535-435a-a61e-b8de03c4e8db" />
The model has been loaded onto the 10th card of 33, but it is displayed as an error in the server worker.

**Comment by gitlawr** [comment]
What is the error message if you move the cursor on it? Is the vllm process still running on your worker?

**Comment by benchiang** [comment]
It will display: Failed to initialize: Timeout waiting for model instance to be initializing.
The vllm process is still running.

**Comment by benchiang** [comment]
Clear the log, restart the service, and run it again, and this error is reported：
<img width="1147" alt="Image" src="https://github.com/user-attachments/assets/5166a2d2-f148-4b87-96cd-9c21441dc6f1" />

<img width="545" alt="Image" src="https://github.com/user-attachments/assets/e32b1485-5b27-441e-b13e-901116f09054" />
The content of /var/lib/gpustack/log/serve/1.log on server 33 is: 2025-02-25T11:27:40+00:00 - gpustack.worker.backends.base - ERROR - Failed to initialize: Timeout waiting for model instance to be initializing.

After deleting and rebuilding, the above error was reported again when running again. After deleting and rebuilding again, the 33 logs showed normal. The attachment is all the log information of the two nodes.

[33-gpustack.log](https://github.com/user-attachments/files/18963397/33-gpustack.log)
[34-gpustack.log](https://github.com/user-attachments/files/18963396/34-gpustack.log)

[33-log.zip](https://github.com/user-attachments/files/18963407/33-log.zip)

[34-log.zip](https://github.com/user-attachments/files/18963410/34-log.zip)

**Comment by benchiang** [comment]
<img width="1078" alt="Image" src="https://github.com/user-attachments/assets/0fa0b9c0-ca8f-482c-8796-10d56f728d95" />

<img width="1046" alt="Image" src="https://github.com/user-attachments/assets/c42be47b-cf41-438c-8330-3dacb9f719e6" />

<img width="1280" alt="Image" src="https://github.com/user-attachments/assets/43e5e4c5-66b2-4dc5-9895-15b674a151e7" />

**Comment by benchiang** [comment]
When solving other problems, I suddenly found that I could communicate normally.
